app/domain/authentication/resource_restrictions/fetch_non_empty_resource_annotations.rb

```diff
@@ -0,0 +1,46 @@
+require 'command_class'
+
+module Authentication
+  module ResourceRestrictions
+
+    FetchNonEmptyResourceAnnotations = CommandClass.new(
+      dependencies: {
+        role_class: ::Role,
+        resource_class: ::Resource,
+      },
+      inputs: %i[account role_name]
+    ) do
+
+      def call
+        fetch_resource_annotations
+      end
+
+      private
+
+      def fetch_resource_annotations
+        resource_annotations
+      end
+
+      def resource_annotations
+        resource.annotations.each_with_object({}) do |annotation, result|
+          annotation_values = annotation.values
+          value = annotation_values[:value]
+          raise Errors::Authentication::ResourceRestrictions::EmptyAnnotationGiven, annotation_values[:name] if value.blank?
```

Comment:
line too long guard imo

```suggestion
          if value.blank?
            raise Errors::Authentication::ResourceRestrictions::EmptyAnnotationGiven, annotation_values[:name]
          end
```

Reply:
also - i understand that this is the only change from the other class. I would leave it as one class and pass a param named `allow_empty_annotations` with default value `true`. this way you can call it with `false` in `authn-jwt`, leave the other calls as-is and it is very clear from reading the code that this is an option and that it allows by default.